server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/luno/LunoExchange.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    @Override
+    public Set<String> getFiatCurrencies() {
+        Set<String> fiatCurrencies = new HashSet<>();
+        fiatCurrencies.add(FiatCurrency.ZAR.getCode());
+        return fiatCurrencies;
+    }
+
+    @Override
+    public BigDecimal getCryptoBalance(String cryptoCurrency) {
+        final LunoBalanceData balance = api.getBalance();
+        final BigDecimal cryptoballance = balance.getBalance("XBT");
+        log.debug("{} exbalance = {}", cryptoCurrency, cryptoballance);
+        return cryptoballance;
+    }
+
+    @Override
+    public String getDepositAddress(String cryptoCurrency) {
+        if (cryptoCurrency.equals("BTC")) {
+            final LunoAddressData address = api.getAddress("XBT");
+            return address.getAddress();
+        } else {
+            final LunoAddressData address = api.getAddress(cryptoCurrency);
+            return address.getAddress();
+        }
+    }
+
+    @Override
+    public String getPreferredFiatCurrency() {
+        return this.preferredFiatCurrency;
+    }
+    
+    
+    @Override
+    public BigDecimal getFiatBalance(String fiatCurrency) {
+        final LunoBalanceData balance = api.getBalance();
+        final BigDecimal fiatballance = balance.getBalance("ZAR");
+        log.debug("{} exbalance = {}", fiatCurrency, fiatballance);
+        return fiatballance;
+    }
+    
+    
+    @Override
+    public String purchaseCoins(BigDecimal amount, String cryptoCurrency, String fiatCurrencyToUse, String description) {
+        String type = "BUY";
+        String pair = "XBTZAR";
+        final LunoTickerData btcZar = api.getTicker("XBTZAR");
+        BigDecimal pricebid  = btcZar.getBid();
+        BigDecimal one       = new BigDecimal(1);
+        BigDecimal onepr     = new BigDecimal(1.01);
+        BigDecimal btcfee    = new BigDecimal(0.00033);
+        amount               = amount.multiply(onepr);
+        amount               = amount.add(btcfee).setScale(6, BigDecimal.ROUND_CEILING);
+        BigDecimal price     = pricebid.add(one).setScale(0, BigDecimal.ROUND_CEILING);
+        BigDecimal amountbtc = price.multiply(amount).setScale(2, BigDecimal.ROUND_CEILING);
+        if (this.typeorder.equals("limit")) {
+            log.debug("limit pair {} type {} amount {} price {}", pair, "BID", amount.toString(), price.toString());
+            final LunoOrderData result = api.createLimitBuyOrder(pair, "BID", amount.toString(), price.toString());
+            return result.getResult();
+        } else {
+            log.debug("market pair {} type {} amount   {}  ", pair, type, amountbtc.toString());
+            final LunoOrderData result = api.createBuyOrder(pair, type, amountbtc.toString());
+            return result.getResult();
+        }
+    }
+    
+    
+    @Override
+    public String sellCoins(BigDecimal cryptoAmount, String cryptoCurrency, String fiatCurrencyToUse, String description) {
+        String type = "SELL";
+        String pair = "XBTZAR";
```

Comment:
Build currency pair based on cryptoCurrency, and fiatCurrencyToUse. Remove hardcoded XBT and ZAR.